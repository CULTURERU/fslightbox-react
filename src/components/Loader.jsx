import React from 'react';

const Loader = () =>
    (
        <div className="fslightbox-loader">
            <div></div>
            <div></div>
            <div></div>
            <div></div>
        </div>
    );

export default Loader;
