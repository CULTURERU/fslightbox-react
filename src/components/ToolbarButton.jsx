import React, { Component } from 'react';

class ToolbarButton extends Component {
    render() {
        return (
            <div>
                xd
            </div>
        );
    }
}

export default ToolbarButton;