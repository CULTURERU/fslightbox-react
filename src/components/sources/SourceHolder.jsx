import React, { Component } from 'react';
import Source from "./Source.jsx";
import PropTypes from 'prop-types';
import SourceTypeChecker from "../../core/Source/SourceTypeChecker";

class SourceHolder extends Component {

    constructor(props) {
        super(props);
        this.source = React.createRef();

        if (!this.props.fsLightbox.sourcesTypes[this.props.index])
            this.initRequest();
        // component need to be mounted to call method from child by ref
        this._isMounted = false;
        this._isTypeChecked = false;
    }

    initRequest() {
        this.sourceTypeChecker = new SourceTypeChecker();
        this.sourceTypeChecker.setUrlToCheck(this.props.fsLightbox.urls[this.props.index]);
        this.sourceTypeChecker.getSourceType()
            .then(() => this.processReceivedSourceType());
    }

    processReceivedSourceType() {
        this.props.fsLightbox.sourcesTypes[this.props.index] = this.sourceTypeChecker.sourceType;
        if (this._isMounted) {
            if (this.source.current === null) {
                this.props.fsLightbox.sourcesToCreateOnConstruct[this.props.index] = true;
                return;
            }
            this.source.current.createSource();
        } else {
            this._isTypeChecked = true;
        }
    }

    componentDidMount() {
        this._isMounted = true;
        if (this._isTypeChecked) {
            this.source.current.createSource();
        }
    }

    render() {
        return (
            <div ref={ this.props.fsLightbox.elements.sourceHolders[this.props.index] }
                 className="fslightbox-source-holder">
                <Source
                    fsLightbox={ this.props.fsLightbox }
                    index={ this.props.index }
                    ref={ this.source }
                />
            </div>
        );
    }
}

SourceHolder.propTypes = {
    fsLightbox: PropTypes.object,
    index: PropTypes.number
};
export default SourceHolder;