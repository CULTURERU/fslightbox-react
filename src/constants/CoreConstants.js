export const CONTAINER_FADE_OUT_TIME = 250;
export const IMAGE_TYPE = 'image';
export const VIDEO_TYPE = 'video';
export const YOUTUBE_TYPE  = 'youtube';
export const INVALID_TYPE  = 'invalid';
