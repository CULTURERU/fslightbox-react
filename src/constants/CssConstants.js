export const FADE_IN_CLASS_NAME = 'fslightbox-fade-in';
export const FADE_OUT_CLASS_NAME = 'fslightbox-fade-out';
export const FADE_IN_COMPLETE_CLASS_NAME = 'fslightbox-fade-in-complete';
export const FADE_OUT_COMPLETE_CLASS_NAME = 'fslightbox-fade-out-complete';
export const FSLIGHTBOX_OPEN_CLASS_NAME = 'fslightbox-open';
export const CURSOR_GRABBING_CLASS_NAME = 'fslightbox-cursor-grabbing';
export const FADE_IN_ANIMATION_TIME = 250;
export const TRANSFORM_TRANSITION_CLASS_NAME = 'fslightbox-transform-transition';