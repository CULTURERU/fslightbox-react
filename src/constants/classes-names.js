export const PREFIX = 'fslightbox-';

// single classes names
export const CURSOR_GRABBING_CLASS_NAME = `${ PREFIX }cursor-grabbing`;
export const FADE_IN_CLASS_NAME = `${ PREFIX }fade-in`;
export const FADE_OUT_CLASS_NAME = `${ PREFIX }fade-out`;
export const FULL_DIMENSION_CLASS_NAME = `${ PREFIX }full-dimension`;
export const FLEX_CENTERED_CLASS_NAME = `${ PREFIX }flex-centered`;
export const FADE_IN_LONG_CLASS_NAME = `${ PREFIX }fade-in-long`;
export const LONG_FADE_OUT_CLASS_NAME = `${ PREFIX }fade-out-long`;
export const OPACITY_0_CLASS_NAME = `${ PREFIX }opacity-0`;
export const OPEN_CLASS_NAME = `${ PREFIX }open`;
export const SLIDE_BTN_CONTAINER_CLASS_NAME = `${ PREFIX }slide-btn-container`;
export const TRANSFORM_TRANSITION_CLASS_NAME = `${ PREFIX }transform-transition`;

// multiple classes names in one string
export const SOURCE_CLASSES_NAMES = `${ PREFIX }source ${ OPACITY_0_CLASS_NAME }`;
