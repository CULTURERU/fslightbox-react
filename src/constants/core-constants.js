// sources types
export const IMAGE_TYPE = 'image';
export const VIDEO_TYPE = 'video';
export const YOUTUBE_TYPE = 'youtube';
export const INVALID_TYPE = 'invalid';
