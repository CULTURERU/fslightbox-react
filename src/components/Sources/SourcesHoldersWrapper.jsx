import React from 'react';
import PropTypes from 'prop-types';
import SourceHolder from "./SourceHolder.jsx";

const SourcesHoldersWrapper = ({ fsLightbox }) => {
    const {
        data: { totalSlides },
        elements: {
            mediaHolder
        },
        core: {
            slideSwiping: {
                down: {
                    listener
                }
            }
        },
    } = fsLightbox;

    const sourceHolders = [];
    for (let i = 0; i < totalSlides; i++) {
        sourceHolders.push(
            <SourceHolder
                fsLightbox={ fsLightbox }
                index={ i }
                key={ i }
            />
        );
    }

    return (
        <div className={ 'fslightbox-media-holder' }
             onMouseDown={ listener }
             onTouchStart={ listener }
             ref={ mediaHolder }>
            { sourceHolders }
        </div>
    );
};

SourcesHoldersWrapper.propTypes = {
    fsLightbox: PropTypes.object.isRequired,
};

export default SourcesHoldersWrapper;