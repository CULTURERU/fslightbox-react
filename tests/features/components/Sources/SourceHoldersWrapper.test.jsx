import React from 'react';
import { mount, shallow } from 'enzyme/build';
import SourcesHoldersWrapper from "../../../../src/components/Sources/SourcesHoldersWrapper";
import { FsLightboxMock } from "../../../__mocks__/components/fsLightboxMock";
import { IMAGE_TYPE } from "../../../../src/constants/CoreConstants";
import SourceHolder from "../../../../src/components/Sources/SourceHolder";

let fsLightbox = {
    data: {
        totalSlides: 4
    },
    elements: {
        sourcesHoldersWrapper: {
            current: null
        }
    },
    core: {
        slideSwiping: {
            down: jest.fn()
        }
    }
};
let sourcesHoldersWrapper;

describe('attaching element to sourcesWrapper ref from fsLightbox elements object', () => {
    let completeFsLightboxObject;

    beforeAll(() => {
        const fsLightboxMock = new FsLightboxMock();
        completeFsLightboxObject = fsLightboxMock.getFsLightbox();
        // as we are using mount
        // we mock sourcesTypes to not load sources via Xhr, because that would be waste of performance
        completeFsLightboxObject.sourcesData.sourcesTypes = [IMAGE_TYPE, IMAGE_TYPE, IMAGE_TYPE, IMAGE_TYPE];
        sourcesHoldersWrapper = mount(<SourcesHoldersWrapper fsLightbox={ completeFsLightboxObject }/>);
    });

    it('should attach element to ref', () => {
        expect(completeFsLightboxObject.elements.sourcesWrapper.current).toEqual(sourcesHoldersWrapper.getDOMNode());
    });
});

describe('MediaHolder DOM', () => {
    beforeAll(() => {
        sourcesHoldersWrapper = shallow(<SourcesHoldersWrapper fsLightbox={ fsLightbox }/>)
    });

    it('should match snapshot', () => {
        expect(sourcesHoldersWrapper).toMatchSnapshot();
    });
});


describe('calling on mouseDown and touchStart events', () => {
    beforeEach(() => {
        fsLightbox.core.slideSwiping.down.listener = jest.fn();
        sourcesHoldersWrapper = shallow(<SourcesHoldersWrapper fsLightbox={ fsLightbox }/>);
    });

    describe('onMouseDown', () => {
        beforeEach(() => {
            sourcesHoldersWrapper.simulate('mouseDown');
        });

        it('should call down listener', () => {
            expect(fsLightbox.core.slideSwiping.down.listener).toBeCalled();
        });
    });


    describe('onTouchStart', () => {
        beforeEach(() => {
            sourcesHoldersWrapper.simulate('touchStart');
        });

        it('should call down listener', () => {
            expect(fsLightbox.core.slideSwiping.down.listener).toBeCalled();
        });
    });
});

describe('rendering source holders', () => {
    beforeAll(() => {
        sourcesHoldersWrapper = shallow(<SourcesHoldersWrapper fsLightbox={ fsLightbox }/>)
    });

    describe('rendering equivalent to totalSlides number of source holders', () => {
        it('should render 4 source holders', () => {
            expect(sourcesHoldersWrapper.find('SourceHolder').length).toEqual(4);
        });
    });

    describe('SourceHolders (testing correct props and dom)', () => {
        for (let i = 0; i < fsLightbox.data.totalSlides; i++) {
            let sourceHolder;
            beforeAll(() => {
                sourceHolder = sourcesHoldersWrapper.childAt(i);
            });

            it(`should be equal to SourceHolder with key and index equals to iteration number 
                and fsLightbox passed`, () => {
                expect(sourceHolder.equals(
                    <SourceHolder
                        fsLightbox={ fsLightbox }
                        index={ i }
                        key={ i }
                    />
                )).toBeTruthy();
            });
        }
    });
});

