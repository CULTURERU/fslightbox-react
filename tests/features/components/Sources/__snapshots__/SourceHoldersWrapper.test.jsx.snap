// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaHolder DOM should match snapshot 1`] = `
<div
  className="fslightbox-media-holder"
>
  <SourceHolder
    fsLightbox={
      Object {
        "core": Object {
          "slideSwiping": Object {
            "down": [MockFunction],
          },
        },
        "data": Object {
          "totalSlides": 4,
        },
        "elements": Object {
          "sourcesHoldersWrapper": Object {
            "current": null,
          },
        },
      }
    }
    index={0}
    key="0"
  />
  <SourceHolder
    fsLightbox={
      Object {
        "core": Object {
          "slideSwiping": Object {
            "down": [MockFunction],
          },
        },
        "data": Object {
          "totalSlides": 4,
        },
        "elements": Object {
          "sourcesHoldersWrapper": Object {
            "current": null,
          },
        },
      }
    }
    index={1}
    key="1"
  />
  <SourceHolder
    fsLightbox={
      Object {
        "core": Object {
          "slideSwiping": Object {
            "down": [MockFunction],
          },
        },
        "data": Object {
          "totalSlides": 4,
        },
        "elements": Object {
          "sourcesHoldersWrapper": Object {
            "current": null,
          },
        },
      }
    }
    index={2}
    key="2"
  />
  <SourceHolder
    fsLightbox={
      Object {
        "core": Object {
          "slideSwiping": Object {
            "down": [MockFunction],
          },
        },
        "data": Object {
          "totalSlides": 4,
        },
        "elements": Object {
          "sourcesHoldersWrapper": Object {
            "current": null,
          },
        },
      }
    }
    index={3}
    key="3"
  />
</div>
`;
